packages/flutter_tools/test/general.shard/android/gradle_errors_test.dart

```diff
@@ -1303,6 +1304,58 @@ at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)'''
       BotDetector: () => const FakeBotDetector(false),
     });
   });
+
+  group('incompatible java and gradle versions error', () {
+    const String errorMessage = '''
+Could not compile build file '…/example/android/build.gradle'.
+> startup failed:
+  General error during conversion: Unsupported class file major version 61
+  java.lang.IllegalArgumentException: Unsupported class file major version 61
+''';
+
+    testWithoutContext('pattern', () {
+      expect(
+        incompatibleJavaAndGradleVersionsHandler.test(errorMessage),
+        isTrue,
+      );
+    });
+
+    testUsingContext('suggestion', () async {
+      await incompatibleJavaAndGradleVersionsHandler.handler(
+        line: errorMessage,
+        project: FlutterProject.fromDirectoryTest(fileSystem.currentDirectory),
+        usesAndroidX: true,
+        multidexEnabled: true,
+      );
+
+      expect(
+        testLogger.statusText,
+        contains(
+          '\n'
+          '┌─ Flutter Fix ────────────────────────────────────────────────────────────────────────────────────┐\n'
+          '│ [!] Your project\'s Gradle and AGP version is incompatible with the Java version that Flutter     │\n'
```

Comment:
Cannot explain this spacing. Not sure why it's off by one space.

Reply:
It seems like these test check the full output but fwiw I dont think that is required. If you are having a spacing issue verify the message contains the parts you want. 
```
expect(testLogger.statusText, contains(RegExp(r'Gradle and AGP version is incompatible with the Java version')));
expect(testLogger.statusText, contains(RegExp(r'https://docs.gradle.org/current/userguide/compatibility.html#java')));
expect(testLogger.statusText, contains(RegExp(r'https://developer.android.com/studio/releases/gradle-plugin#updating-gradle')));
```

Another option would be to make a method that formats this string visibleForTesting and use that method to ensure the data you expect is present.